File: VSCode/Python/Random/binarySearch.py
array = [1,2,3,4,5]

x = input("Enter the search term: ")
x = int(x)


def binarySearch():
    start = 0
    end = len(array) - 1

    while start <= end:
        middle = (start + end) // 2
        if x == array[middle] :
            return "found at index "+str(middle)
            exit()
        elif x > array[middle]:
            start = middle + 1
        elif x < array[middle]:
            end = middle - 1
    return "not found"

y= binarySearch()
print("The search term was "+y)

File: Binary_Search/BinarySearch.rb
$array = Array[1,2,3,4,5]

puts "Enter the search term: "
x = gets.to_i

def binarySearch(x)
    a, b = 0, $array.length - 1
    while a <= b do
        middle = (a + b) / 2
        if x == $array[middle]
            return middle
        elsif x > $array[middle]
            a = middle + 1
        elsif x < $array[middle]
            b = middle - 1
        end
    end
    return -1
end

y = binarySearch(x)
if y != -1
    puts "The search term was found at index "+ y.to_s
else
    puts "The search term was not found"
end
